packages/react/src/drawer/docs/overview.mdx

```diff
@@ -347,3 +347,70 @@ By default, Drawer’s overlay defaults to 80% opacity, however, by adding `mute
 	);
 };
 ```
+
+## Focus custom element on close
+
+When the drawer closes, focus is returned to the element that triggered its opening. You can override this behaviour by passing a custom element to the `elementToFocusOnClose` prop. This is useful when an action performed in the drawer causes an error and focus should be returned to an alert on close.
+
+```jsx live
+() => {
+	const sectionAlertContainerRef = React.useRef(null);
+	const [error, setError] = React.useState('');
+	const [isDrawerOpen, openDrawer, closeDrawer] = useTernaryState(false);
+
+	const openDrawerAndClearError = () => {
+		openDrawer();
+		setError('');
+	};
+
+	const closeWithError = () => {
+		closeDrawer();
+		setError('We were unable to perform the action, please try again later.');
+	};
+
+	return (
+		<Stack gap={2}>
+			<Box
+				css={error ? undefined : { display: 'none' }}
+				focusRingFor="all"
+				ref={sectionAlertContainerRef}
+				tabIndex={-1}
```

Comment:
None of this should exist?
```suggestion
```

Reply:
Yeah, all this is needed. It's not needed on the alert, though. Drawer is pernickety.

Reply:
Isn't it better to have it on the alert instead of this outer element? It seems to still function

Reply:
Hmm it didn't in my testing. Did you edit the docs or the code?

There's no `ref.current` when the drawer opens if you conditionally render the alert. `ref` updates don't trigger component re-renders so the `elementToFocusOnClose` prop never updates and no focus is set via the drawer.

Reply:
It should be this:

```
<Box css={error ? undefined : { display: 'none' }}>
	<SectionAlert
		title="Something went wrong"
		tone="error" focusRingFor="all"
		ref={sectionAlertContainerRef}
		tabIndex={-1}
	>
		<Text as="p">{error}</Text>
	</SectionAlert>
</Box>
```
			
Otherwise no focus ring will be applied since `tabIndex={-1}` causes the focus ring for all on `SectionAlert`, but not for `Box`

Reply:
Actually, your solution should work too, but doesn't…? Why isn't the focus ring being applied? I'd still prefer my solution since we should encourage focusing the actual alert, but we need to figure out why yours doesn't do anything